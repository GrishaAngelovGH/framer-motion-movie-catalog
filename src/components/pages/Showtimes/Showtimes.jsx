import { useState } from "react"

import LayoutPage from "components/pages/LayoutPage"
import CinemaModal from "components/CinemaModal"

const Showtimes = () => {
  const [cinemas, setCinemas] = useState([])
  const [showModal, setShowModal] = useState(false)

  const toggleModal = () => {
    !setShowModal(!showModal)
  }

  return (
    <LayoutPage>
      <div className="bg-blue-800 min-h-full flex flex-col items-center">
        {showModal && (<CinemaModal onClose={toggleModal} />)}

        <div className="m-10 md:w-1/2">
          <div className="bg-blue-900 text-white rounded-t-md p-3">
            Show me times for Cinemas
          </div>
          {
            !cinemas.length && (
              <div className="border-2 border-blue-900 border-t-0 rounded-b-md p-5 flex flex-col items-center text-white text-2xl text-center">
                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="w-[200px] h-[200px] text-blue-500">
                  <path strokeLinecap="round" strokeLinejoin="round" d="M3.375 19.5h17.25m-17.25 0a1.125 1.125 0 0 1-1.125-1.125M3.375 19.5h1.5C5.496 19.5 6 18.996 6 18.375m-3.75 0V5.625m0 12.75v-1.5c0-.621.504-1.125 1.125-1.125m18.375 2.625V5.625m0 12.75c0 .621-.504 1.125-1.125 1.125m1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125m0 3.75h-1.5A1.125 1.125 0 0 1 18 18.375M20.625 4.5H3.375m17.25 0c.621 0 1.125.504 1.125 1.125M20.625 4.5h-1.5C18.504 4.5 18 5.004 18 5.625m3.75 0v1.5c0 .621-.504 1.125-1.125 1.125M3.375 4.5c-.621 0-1.125.504-1.125 1.125M3.375 4.5h1.5C5.496 4.5 6 5.004 6 5.625m-3.75 0v1.5c0 .621.504 1.125 1.125 1.125m0 0h1.5m-1.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125m1.5-3.75C5.496 8.25 6 7.746 6 7.125v-1.5M4.875 8.25C5.496 8.25 6 8.754 6 9.375v1.5m0-5.25v5.25m0-5.25C6 5.004 6.504 4.5 7.125 4.5h9.75c.621 0 1.125.504 1.125 1.125m1.125 2.625h1.5m-1.5 0A1.125 1.125 0 0 1 18 7.125v-1.5m1.125 2.625c-.621 0-1.125.504-1.125 1.125v1.5m2.625-2.625c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125M18 5.625v5.25M7.125 12h9.75m-9.75 0A1.125 1.125 0 0 1 6 10.875M7.125 12C6.504 12 6 12.504 6 13.125m0-2.25C6 11.496 5.496 12 4.875 12M18 10.875c0 .621-.504 1.125-1.125 1.125M18 10.875c0 .621.504 1.125 1.125 1.125m-2.25 0c.621 0 1.125.504 1.125 1.125m-12 5.25v-5.25m0 5.25c0 .621.504 1.125 1.125 1.125h9.75c.621 0 1.125-.504 1.125-1.125m-12 0v-1.5c0-.621-.504-1.125-1.125-1.125M18 18.375v-5.25m0 5.25v-1.5c0-.621.504-1.125 1.125-1.125M18 13.125v1.5c0 .621.504 1.125 1.125 1.125M18 13.125c0-.621.504-1.125 1.125-1.125M6 13.125v1.5c0 .621-.504 1.125-1.125 1.125M6 13.125C6 12.504 5.496 12 4.875 12m-1.5 0h1.5m-1.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125M19.125 12h1.5m0 0c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125m-17.25 0h1.5m14.25 0h1.5" />
                </svg>
                <h3 className="m-1">Please choose a cinema</h3>
                <h3 className="m-1">Where would you like to see the film?</h3>
                <button
                  onClick={toggleModal}
                  className="mt-10 bg-slate-400 hover:bg-slate-300 text-gray-600"
                >
                  Add Cinemas
                </button>
              </div>
            )
          }
        </div>
      </div>
    </LayoutPage>
  )
}

export default Showtimes