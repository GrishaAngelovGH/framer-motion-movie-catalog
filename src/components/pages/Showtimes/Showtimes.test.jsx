import { render } from "@testing-library/react"
import { BrowserRouter as Router, Routes, Route } from "react-router-dom"

import Showtimes from "./Showtimes"

test("should render pages/Showtimes component", () => {
  const view = render(
    <Router>
      <Routes>
        <Route path="/" element={<Showtimes />} />
      </Routes>
    </Router>
  )

  expect(view).toMatchSnapshot()
})