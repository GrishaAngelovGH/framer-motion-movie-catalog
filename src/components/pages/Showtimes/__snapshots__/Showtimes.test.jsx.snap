// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render pages/Showtimes component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col h-screen"
      >
        <div
          class="lg:h-[60px]"
        >
          <div
            class="flex flex-col md:flex-row justify-between items-center bg-blue-500 h-full p-2"
          >
            <a
              class="ms-5 flex items-center"
              href="/"
            >
              <img
                class="w-[40px]"
                src="/images/film-roll.png"
              />
              <h1
                class="ms-3 text-2xl font-bold text-blue-900"
              >
                Cinema
              </h1>
            </a>
            <div
              class="m-5 grid grid-rows-2 grid-cols-2 gap-2 md:grid-flow-col md:grid-rows-1"
            >
              <a
                class="text-white text-center hover:text-white hover:underline me-5 bg-blue-800 p-2 rounded-full"
                href="/"
              >
                Home
              </a>
              <a
                class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
                href="/favorites"
              >
                Favorites
              </a>
              <a
                class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
                href="/experiences"
              >
                Experiences
              </a>
            </div>
          </div>
        </div>
        <div
          class="h-full"
        >
          <div
            class="bg-blue-800 min-h-full flex flex-col items-center"
          >
            <div
              class="m-10 md:w-1/2"
            >
              <div
                class="bg-blue-900 text-white rounded-t-md p-3"
              >
                Show me times for 
                <span
                  class="text-blue-500 hover:text-blue-400 hover:cursor-pointer"
                >
                  Cinemas
                </span>
              </div>
              <div
                class="border-2 border-blue-900 border-t-0 rounded-b-md p-5 flex flex-col items-center text-white text-2xl text-center"
              >
                <svg
                  class="w-[200px] h-[200px] text-blue-500"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.375 19.5h17.25m-17.25 0a1.125 1.125 0 0 1-1.125-1.125M3.375 19.5h1.5C5.496 19.5 6 18.996 6 18.375m-3.75 0V5.625m0 12.75v-1.5c0-.621.504-1.125 1.125-1.125m18.375 2.625V5.625m0 12.75c0 .621-.504 1.125-1.125 1.125m1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125m0 3.75h-1.5A1.125 1.125 0 0 1 18 18.375M20.625 4.5H3.375m17.25 0c.621 0 1.125.504 1.125 1.125M20.625 4.5h-1.5C18.504 4.5 18 5.004 18 5.625m3.75 0v1.5c0 .621-.504 1.125-1.125 1.125M3.375 4.5c-.621 0-1.125.504-1.125 1.125M3.375 4.5h1.5C5.496 4.5 6 5.004 6 5.625m-3.75 0v1.5c0 .621.504 1.125 1.125 1.125m0 0h1.5m-1.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125m1.5-3.75C5.496 8.25 6 7.746 6 7.125v-1.5M4.875 8.25C5.496 8.25 6 8.754 6 9.375v1.5m0-5.25v5.25m0-5.25C6 5.004 6.504 4.5 7.125 4.5h9.75c.621 0 1.125.504 1.125 1.125m1.125 2.625h1.5m-1.5 0A1.125 1.125 0 0 1 18 7.125v-1.5m1.125 2.625c-.621 0-1.125.504-1.125 1.125v1.5m2.625-2.625c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125M18 5.625v5.25M7.125 12h9.75m-9.75 0A1.125 1.125 0 0 1 6 10.875M7.125 12C6.504 12 6 12.504 6 13.125m0-2.25C6 11.496 5.496 12 4.875 12M18 10.875c0 .621-.504 1.125-1.125 1.125M18 10.875c0 .621.504 1.125 1.125 1.125m-2.25 0c.621 0 1.125.504 1.125 1.125m-12 5.25v-5.25m0 5.25c0 .621.504 1.125 1.125 1.125h9.75c.621 0 1.125-.504 1.125-1.125m-12 0v-1.5c0-.621-.504-1.125-1.125-1.125M18 18.375v-5.25m0 5.25v-1.5c0-.621.504-1.125 1.125-1.125M18 13.125v1.5c0 .621.504 1.125 1.125 1.125M18 13.125c0-.621.504-1.125 1.125-1.125M6 13.125v1.5c0 .621-.504 1.125-1.125 1.125M6 13.125C6 12.504 5.496 12 4.875 12m-1.5 0h1.5m-1.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125M19.125 12h1.5m0 0c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125m-17.25 0h1.5m14.25 0h1.5"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <h3
                  class="m-1"
                >
                  Please choose a cinema
                </h3>
                <h3
                  class="m-1"
                >
                  Where would you like to see the film?
                </h3>
                <button
                  class="mt-10 bg-slate-400 hover:bg-slate-300 text-gray-600"
                >
                  Add Cinemas
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col h-screen"
    >
      <div
        class="lg:h-[60px]"
      >
        <div
          class="flex flex-col md:flex-row justify-between items-center bg-blue-500 h-full p-2"
        >
          <a
            class="ms-5 flex items-center"
            href="/"
          >
            <img
              class="w-[40px]"
              src="/images/film-roll.png"
            />
            <h1
              class="ms-3 text-2xl font-bold text-blue-900"
            >
              Cinema
            </h1>
          </a>
          <div
            class="m-5 grid grid-rows-2 grid-cols-2 gap-2 md:grid-flow-col md:grid-rows-1"
          >
            <a
              class="text-white text-center hover:text-white hover:underline me-5 bg-blue-800 p-2 rounded-full"
              href="/"
            >
              Home
            </a>
            <a
              class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
              href="/favorites"
            >
              Favorites
            </a>
            <a
              class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
              href="/experiences"
            >
              Experiences
            </a>
          </div>
        </div>
      </div>
      <div
        class="h-full"
      >
        <div
          class="bg-blue-800 min-h-full flex flex-col items-center"
        >
          <div
            class="m-10 md:w-1/2"
          >
            <div
              class="bg-blue-900 text-white rounded-t-md p-3"
            >
              Show me times for 
              <span
                class="text-blue-500 hover:text-blue-400 hover:cursor-pointer"
              >
                Cinemas
              </span>
            </div>
            <div
              class="border-2 border-blue-900 border-t-0 rounded-b-md p-5 flex flex-col items-center text-white text-2xl text-center"
            >
              <svg
                class="w-[200px] h-[200px] text-blue-500"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.375 19.5h17.25m-17.25 0a1.125 1.125 0 0 1-1.125-1.125M3.375 19.5h1.5C5.496 19.5 6 18.996 6 18.375m-3.75 0V5.625m0 12.75v-1.5c0-.621.504-1.125 1.125-1.125m18.375 2.625V5.625m0 12.75c0 .621-.504 1.125-1.125 1.125m1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125m0 3.75h-1.5A1.125 1.125 0 0 1 18 18.375M20.625 4.5H3.375m17.25 0c.621 0 1.125.504 1.125 1.125M20.625 4.5h-1.5C18.504 4.5 18 5.004 18 5.625m3.75 0v1.5c0 .621-.504 1.125-1.125 1.125M3.375 4.5c-.621 0-1.125.504-1.125 1.125M3.375 4.5h1.5C5.496 4.5 6 5.004 6 5.625m-3.75 0v1.5c0 .621.504 1.125 1.125 1.125m0 0h1.5m-1.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125m1.5-3.75C5.496 8.25 6 7.746 6 7.125v-1.5M4.875 8.25C5.496 8.25 6 8.754 6 9.375v1.5m0-5.25v5.25m0-5.25C6 5.004 6.504 4.5 7.125 4.5h9.75c.621 0 1.125.504 1.125 1.125m1.125 2.625h1.5m-1.5 0A1.125 1.125 0 0 1 18 7.125v-1.5m1.125 2.625c-.621 0-1.125.504-1.125 1.125v1.5m2.625-2.625c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125M18 5.625v5.25M7.125 12h9.75m-9.75 0A1.125 1.125 0 0 1 6 10.875M7.125 12C6.504 12 6 12.504 6 13.125m0-2.25C6 11.496 5.496 12 4.875 12M18 10.875c0 .621-.504 1.125-1.125 1.125M18 10.875c0 .621.504 1.125 1.125 1.125m-2.25 0c.621 0 1.125.504 1.125 1.125m-12 5.25v-5.25m0 5.25c0 .621.504 1.125 1.125 1.125h9.75c.621 0 1.125-.504 1.125-1.125m-12 0v-1.5c0-.621-.504-1.125-1.125-1.125M18 18.375v-5.25m0 5.25v-1.5c0-.621.504-1.125 1.125-1.125M18 13.125v1.5c0 .621.504 1.125 1.125 1.125M18 13.125c0-.621.504-1.125 1.125-1.125M6 13.125v1.5c0 .621-.504 1.125-1.125 1.125M6 13.125C6 12.504 5.496 12 4.875 12m-1.5 0h1.5m-1.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125M19.125 12h1.5m0 0c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125m-17.25 0h1.5m14.25 0h1.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <h3
                class="m-1"
              >
                Please choose a cinema
              </h3>
              <h3
                class="m-1"
              >
                Where would you like to see the film?
              </h3>
              <button
                class="mt-10 bg-slate-400 hover:bg-slate-300 text-gray-600"
              >
                Add Cinemas
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render pages/Showtimes component with selected cinemas 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col h-screen"
      >
        <div
          class="lg:h-[60px]"
        >
          <div
            class="flex flex-col md:flex-row justify-between items-center bg-blue-500 h-full p-2"
          >
            <a
              class="ms-5 flex items-center"
              href="/"
            >
              <img
                class="w-[40px]"
                src="/images/film-roll.png"
              />
              <h1
                class="ms-3 text-2xl font-bold text-blue-900"
              >
                Cinema
              </h1>
            </a>
            <div
              class="m-5 grid grid-rows-2 grid-cols-2 gap-2 md:grid-flow-col md:grid-rows-1"
            >
              <a
                class="text-white text-center hover:text-white hover:underline me-5 bg-blue-800 p-2 rounded-full"
                href="/"
              >
                Home
              </a>
              <a
                class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
                href="/favorites"
              >
                Favorites
              </a>
              <a
                class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
                href="/experiences"
              >
                Experiences
              </a>
            </div>
          </div>
        </div>
        <div
          class="h-full"
        >
          <div
            class="bg-blue-800 min-h-full flex flex-col items-center"
          >
            <div
              class="m-10 md:w-1/2"
            >
              <div
                class="bg-blue-900 text-white rounded-t-md p-3"
              >
                Show me times for 
                <span
                  class="text-blue-500 hover:text-blue-400 hover:cursor-pointer"
                >
                  Cinemas
                </span>
              </div>
              <div
                class="text-white border rounded-md m-2 p-2"
              >
                <div>
                  Acton
                </div>
                <div
                  class="mb-2"
                >
                  Royale Leisure Park, Kendal Avenue, London
                </div>
                <div
                  class="m-2"
                >
                  <a
                    class="mb-2 bg-blue-500 text-white hover:text-white p-2 rounded-md inline-block"
                    href="/movies/1"
                  >
                    The Garfield Movie
                  </a>
                  <div
                    class="ms-2"
                  >
                    Start Time: 
                    09:15
                  </div>
                  <div
                    class="ms-2"
                  >
                    Screen Name: 
                    Screen 1
                  </div>
                </div>
                <div
                  class="m-2"
                >
                  <a
                    class="mb-2 bg-blue-500 text-white hover:text-white p-2 rounded-md inline-block"
                    href="/movies/2"
                  >
                    Fly Me to the Moon
                  </a>
                  <div
                    class="ms-2"
                  >
                    Start Time: 
                    10:15
                  </div>
                  <div
                    class="ms-2"
                  >
                    Screen Name: 
                    Screen 2
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col h-screen"
    >
      <div
        class="lg:h-[60px]"
      >
        <div
          class="flex flex-col md:flex-row justify-between items-center bg-blue-500 h-full p-2"
        >
          <a
            class="ms-5 flex items-center"
            href="/"
          >
            <img
              class="w-[40px]"
              src="/images/film-roll.png"
            />
            <h1
              class="ms-3 text-2xl font-bold text-blue-900"
            >
              Cinema
            </h1>
          </a>
          <div
            class="m-5 grid grid-rows-2 grid-cols-2 gap-2 md:grid-flow-col md:grid-rows-1"
          >
            <a
              class="text-white text-center hover:text-white hover:underline me-5 bg-blue-800 p-2 rounded-full"
              href="/"
            >
              Home
            </a>
            <a
              class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
              href="/favorites"
            >
              Favorites
            </a>
            <a
              class="text-white text-center hover:text-white hover:underline me-5 bg-transparent p-2"
              href="/experiences"
            >
              Experiences
            </a>
          </div>
        </div>
      </div>
      <div
        class="h-full"
      >
        <div
          class="bg-blue-800 min-h-full flex flex-col items-center"
        >
          <div
            class="m-10 md:w-1/2"
          >
            <div
              class="bg-blue-900 text-white rounded-t-md p-3"
            >
              Show me times for 
              <span
                class="text-blue-500 hover:text-blue-400 hover:cursor-pointer"
              >
                Cinemas
              </span>
            </div>
            <div
              class="text-white border rounded-md m-2 p-2"
            >
              <div>
                Acton
              </div>
              <div
                class="mb-2"
              >
                Royale Leisure Park, Kendal Avenue, London
              </div>
              <div
                class="m-2"
              >
                <a
                  class="mb-2 bg-blue-500 text-white hover:text-white p-2 rounded-md inline-block"
                  href="/movies/1"
                >
                  The Garfield Movie
                </a>
                <div
                  class="ms-2"
                >
                  Start Time: 
                  09:15
                </div>
                <div
                  class="ms-2"
                >
                  Screen Name: 
                  Screen 1
                </div>
              </div>
              <div
                class="m-2"
              >
                <a
                  class="mb-2 bg-blue-500 text-white hover:text-white p-2 rounded-md inline-block"
                  href="/movies/2"
                >
                  Fly Me to the Moon
                </a>
                <div
                  class="ms-2"
                >
                  Start Time: 
                  10:15
                </div>
                <div
                  class="ms-2"
                >
                  Screen Name: 
                  Screen 2
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
